docs/adr/0001-user-roles.rst

```diff
@@ -0,0 +1,51 @@
+ADR 0001 - User roles
+=====================
+
+Status
+------
+
+Proposed
+
+
+Context
+-------
+
+Depending on what we expect them to do on Marcha, we need to define some user roles.
+
+First we have the people managing a Marcha instance.
+
+Then we have people linking their own website (this website is a "consumer site") to a Marcha instance, to host videos.
+
+Next we have the video authors.
+
+And finally we have the users coming to watch videos.
+
```

Comment:
Yes, names are good. 

I am not sure about what you call the organization manager. What organization? Is an organization what I called a "consumer-site" (could be an edx instance or whatever)? Or one consumer-site could have many organizations?

So in my idea:

- admin: is_staff
- manager: person responsible for the link between a consumer-site and a Marsha instance
- author
- viewer

So not sure where if what you call the "organization manager" fits here, but it may be something I missed :)

Reply:
ConsumerSite : fun-mooc.fr or edx.org
Organization : Polytechnique, Stanford or Peugeot

Admins are working for a website
Managers and authors are working for an organization

Reply:
Ok didn't know about organizations

So let's recap:

### Marsha instance

`User` model:

- `is_staff` field set to `True`

### Consumer site

`ConsumerSite` model:

- `ManyToMany` named `admins to `User` model (we can imagine a user working for more than one consumer site)

### Organization

`Organization` model:

- `ForeignKey` named `consumer_site` to `ConsumerSite` model
- `ManyToMany` named `managers` to `User` model

### Author

`User` model:

- `ManyToMany` named `organizations` to `Organization` model

### Viewer

Nothing special

Reply:
The organization manager could be required to create a "course" in Marsha from the admin and linking "authors" to this course who can then upload video to this course?

Reply:
Do you really want it to be that complicated? Just asking, really. The way I saw things, as soon as a manager allows videos on an organization, all authors could use the feature.

If you still want it, could it be done later?

Reply:
yes we can simplify access rights for the first version.

But I think we can't escape a model "Course" or "PlayList" as it is called on the current VideoFront code base.

One more design decision to make:
Organization has many2many to ConsumerSite (university can publish courses on several sites) or everything is isolated by consumer site and the university on the other site will be a different entity?

Whatever we decide, it must be possible to implement publishing a course to another site...

Reply:
1/ About the playlist, it depends: what is the use case?

If videos are integrated one by one in a course, it's the course that take the role of the playlist. 

For me each video is independent from each other and is viewed on the consumer site via a link/iframe/whatever that integrates it on the site.

Except if you want the concept of playlist visible outside of the consumer site. Is it really needed?

2/ About consumer site isolation: as consumer sites are independent, they will have different ids, and on them authors will too... So I don't really know (for now) how we can do it reliably. For me they are different entities. Any idea how to make this work?